daemon/graphdriver/driver.go

```diff
@@ -142,10 +142,36 @@ func GetDriver(name, home string, options []string) (Driver, error) {
 func New(root string, options []string) (driver Driver, err error) {
 	for _, name := range []string{os.Getenv("DOCKER_DRIVER"), DefaultDriver} {
 		if name != "" {
+			logrus.Infof("[graphdriver] trying provided driver %q", name) // so the logs show specified driver
 			return GetDriver(name, root, options)
 		}
 	}
 
+	// Guess for prior driver
```

Comment:
Let's walk through these cases, because i'm of the opinion that the "state" needed here is in the directories present.

Users on "auto", would not be kicked to devicemapper from AUFS, unless aufstools aren't present. But if they had been using `devicemapper`, pulled images, then installed aufs-tools, yes they would need to recreate/pull images. But there is already logging about that.